### PR TITLE
compact: Add benchmark for #4084

### PR DESCRIPTION
**Pull request by zecke**
* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

### CODE REVIEW

**Review by yeya24 — COMMENTED** [review]
This looks nice! Can you show an example run of the benchmark test?

**Comment by zecke** [comment]
```
goos: darwin
goarch: amd64
pkg: github.com/thanos-io/thanos/pkg/compact
cpu: VirtualApple @ 2.50GHz
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-10-10-8         	     478	   2567498 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-10-20-8         	     480	   2608270 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-10-30-8         	     484	   2619140 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-10-40-8         	     490	   2605465 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-10-50-8         	     478	   2609271 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-10-60-8         	     476	   2615486 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-100-10-8        	      46	  25648431 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-100-20-8        	      93	  12734189 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-100-30-8        	     120	  10084033 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-100-40-8        	     158	   7471222 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-100-50-8        	     241	   5062009 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-100-60-8        	     240	   5060903 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-1000-10-8       	       4	 311204156 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-1000-20-8       	       8	 137990490 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-1000-30-8       	      12	  90377771 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-1000-40-8       	      16	  65423945 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-1000-50-8       	      21	  51229673 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-1000-60-8       	      26	  42566138 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-10000-10-8      	       1	4993933375 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-10000-20-8      	       1	2490431959 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-10000-30-8      	       1	1664484417 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-10000-40-8      	       1	1243212958 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-10000-50-8      	       2	 746272979 ns/op
BenchmarkGatherNoCompactionMarkFilter_Filter/Bench-10000-60-8      	       2	 619553000 ns/op
PASS
ok  	github.com/thanos-io/thanos/pkg/compact	49.791s
```

Larger win for more blocks.

**Review by yeya24 — APPROVED** [review]
LGTM! cc @bwplotka @GiedriusS for another look.

File: pkg/compact/compact_test.go [hunk 1]
```diff
@@ -125,3 +137,40 @@ func TestGroupMaxMinTime(t *testing.T) {
 	testutil.Equals(t, int64(0), g.MinTime())
 	testutil.Equals(t, int64(30), g.MaxTime())
 }
+
+func BenchmarkGatherNoCompactionMarkFilter_Filter(b *testing.B) {
+	ctx := context.TODO()
+	logger := log.NewLogfmtLogger(os.Stderr)
```
Comment (GiedriusS) [inline pkg/compact/compact_test.go:143]: Here we are kind of testing how fast is writing messages to the terminal. Maybe we could do:
```suggestion
	logger := log.NewLogfmtLogger(ioutil.Discard)
```
Here?

**Review by GiedriusS — APPROVED** [review]
Awesome work! Just one suggestion. Let me know if it makes sense :+1:

**Comment by GiedriusS** [comment]
You got tricked by the GitHub UI :joy: could you add a DCO? Let's merge this

**Comment by GiedriusS** [comment]
```
# github.com/thanos-io/thanos/pkg/compact [github.com/thanos-io/thanos/pkg/compact.test]
pkg/compact/compact_test.go:11:2: imported and not used: "os"
pkg/compact/compact_test.go:143:32: undefined: ioutil
```
Could you please fix these?

**Review by GiedriusS — APPROVED** [review]

**Pull request merged**